pkg/platform/model/vcs.go

```diff
@@ -873,6 +882,18 @@ func GetRevertCommit(from, to strfmt.UUID) (*mono_models.Commit, error) {
 	return res.Payload, nil
 }
 
+func RevertCommitWithinHistory(from, to strfmt.UUID) (*mono_models.Commit, error) {
+	ok, err := CommitWithinCommitHistory(from, to)
+	if err != nil {
+		return nil, locale.WrapError(err, "err_revert_commit_within_history_no_info", "Cannot determine if commit being reverted to is within the current commit's history.")
```

Comment:
I don't follow what you mean. If this error is encountered, something like a network connection failed and we are not able to figure this out for the user.

Reply:
I mean that when we are localizing the error for the end-user we should be able to interpret the type of error that happened and give an explicit error message. We explicitly check that the history is in order, so we should explicitly error when it is not.

Reply:
If the commit is not in the history, that message can be very clear to users. If the error comes from the API (transport, incorrect api usage, etc.), we'd have to introduce finer responses for all model functions here: https://github.com/ActiveState/cli/blob/c51f834878cc5479f947371026ba119f51d2c430/pkg/platform/api/api.go#L94-L113

Reply:
My point is that "Cannot determine" should not be anywhere in the error message returned to the user, because we CAN and DID determine it. If you are saying that we simply cannot determine it at this particular stage of the execution then that just means that the error should be localized at a different level.